### PR TITLE
[SPARK-41329][CONNECT] Resolve circular imports in Spark Connect

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to resolve the circular imports workarounds

### Why are the changes needed?

For better readability and maintanence.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

CI in this PR should test it out.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @amaliujia

**Review by dongjoon-hyun — COMMENTED** [review]
Is there any chance to be conflict with #38991? Which one do you want to merge first, @HyukjinKwon ?

**Comment by HyukjinKwon** [comment]
I prefer to merge https://github.com/apache/spark/pull/38991 first but please don't bother. I don't mind resolving conflicts 👍

**Review by grundprinzip — APPROVED** [review]

**Review by amaliujia — APPROVED** [review]
LGTM 

Thank you!

File: python/pyspark/sql/connect/column.py [hunk 1]
```diff
@@ -706,28 +705,30 @@ def substr(self, startPos: Union[int, "Column"], length: Union[int, "Column"]) -
...
             raise TypeError("Unsupported type for substr().")
 
         if isinstance(startPos, int):
-            start_exp = self._lit(startPos)
+            start_exp = lit(startPos)
         else:
             start_exp = startPos
 
-        return F.substr(self, start_exp, length_exp)
+        return F.substr(self, start_exp, length_exp)  # type: ignore[attr-defined]
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/column.py:721]:
```suggestion
        return F.substr(self, start_exp, length_exp)
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Ur, does this pass CI?

**Comment by HyukjinKwon** [comment]
oh yeah. the tests passed but the linter failed. I just removed one ignore in linter that's not used, and I checked it locally.

But sorry I should have waited for the test results. Will make sure on that next time.

**Comment by dongjoon-hyun** [comment]
No problem at all. If you checked locally, that's more than enough. :)

**Pull request closed without merging**